service/invoker/invoker_internal_test.go

```diff
@@ -112,6 +112,22 @@ func TestInvoker_Script(t *testing.T) {
 		assert.Error(t, err)
 	})
 
+	t.Run("handles unavailable block data", func(t *testing.T) {
+		t.Parallel()
+
+		index := mocks.BaselineReader(t)
+		index.LastFunc = func() (uint64, error) {
+			return mocks.GenericHeight - 1, nil
+		}
+
+		invoke := baselineInvoker(t)
+		invoke.index = index
+
+		_, err := invoke.Script(mocks.GenericHeight, mocks.GenericBytes, []cadence.Value{})
+
+		assert.Error(t, err)
```

Comment:
to be a bit more robust you likely want to check for the specific error type/text.

Reply:
Done!